### ISSUE TITLE
Compose should not remove stopped containers on scaling up

### ISSUE THREAD

**Issue by nazar-pc** [docker/compose#2559]
```yml
# docker-compose.yml
busybox:
  image: busybox
```
```bash
$ docker-compose up -d
Creating data_busybox_1
$ docker-compose ps
     Name        Command   State    Ports 
-----------------------------------------
data_busybox_1   sh        Exit 0         
$ docker-compose scale busybox=5
Starting data_busybox_1 ... done
Creating and starting 2 ... done
Creating and starting 3 ... done
Creating and starting 4 ... done
Creating and starting 5 ... done
Removing data_busybox_5 ... done
Removing data_busybox_4 ... done
Removing data_busybox_2 ... done
Removing data_busybox_3 ... done
Removing data_busybox_1 ... done
$ docker-compose ps
Name   Command   State   Ports 
------------------------------
$ docker-compose scale busybox=5
Creating and starting 1 ... done
Creating and starting 2 ... done
Creating and starting 3 ... done
Creating and starting 4 ... done
Creating and starting 5 ... done
Removing data_busybox_3 ... done
Removing data_busybox_2 ... done
Removing data_busybox_5 ... done
Removing data_busybox_4 ... done
Removing data_busybox_1 ... done
$ docker-compose ps
Name   Command   State   Ports 
------------------------------
```

I do not understand why containers are removed, I've specified to scale to 5 instances, but it removed even first one.
I'm aware that containers are stopped, they might be just data containers in Swarm cluster, they should not be removed in such cases!

#### docker version
```
Client:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:20:08 UTC 2015
 OS/Arch:      linux/amd64

Server:
 Version:      1.9.1
 API version:  1.21
 Go version:   go1.4.2
 Git commit:   a34a1d5
 Built:        Fri Nov 20 13:20:08 UTC 2015
 OS/Arch:      linux/amd64
```

#### docker info
```
Containers: 10
Images: 321
Server Version: 1.9.1
Storage Driver: btrfs
 Build Version: Btrfs v4.0
 Library Version: 101
Execution Driver: native-0.2
Logging Driver: json-file
Kernel Version: 4.4.0-rc5-haswell
Operating System: Ubuntu 16.04 (containerized)
CPUs: 8
Total Memory: 15.59 GiB
Name: nazar-pc
ID: ***
WARNING: No swap limit support
```

#### docker-compose version
```
docker-compose version 1.5.2, build 7240ff3
docker-py version: 1.5.0
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1e 11 Feb 2013
```

#### uname -a
```
Linux nazar-pc 4.4.0-rc5-haswell #1 SMP Mon Dec 14 03:22:43 CET 2015 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by aanand** [comment]
Right, this is a race condition caused by the implementation of `scale`: it expects the containers to stay alive, but if they exit immediately then they get "cleaned up". Shouldn't be a difficult fix.

**Comment by dbonev** [comment]
Currently, the way ```scale``` is implemented, every time it scales up or down, it will unconditionally remove all stopped containers from a service (no matter when and how these containers were stopped):

https://github.com/docker/compose/blob/master/compose/service.py#L225

Shouldn't ```scale``` be more conservative when it comes to removal? More precisely shouldn't we remove only those containers stopped during a ```scale down``` operation?
I.e. something like this maybe:
```diff
--- a/compose/service.py
+++ b/compose/service.py
@@ -218,11 +218,11 @@ class Service(object):
             sorted_running_containers = sorted(
                 running_containers,
                 key=attrgetter('number'))
+            containers_to_stop = sorted_running_containers[-num_to_stop:]
             parallel_stop(
-                sorted_running_containers[-num_to_stop:],
+                containers_to_stop,
                 dict(timeout=timeout))
-
-        parallel_remove(self.containers(stopped=True), {})
+            parallel_remove(containers_to_stop, {})

```

**Comment by nazar-pc** [comment]
This issue is targeted to 1.6.0, but rc1 doesn't seem to contain fix.
Is there still a chance for this to be fixed before release?

**Comment by aanand** [comment]
Yes, it'll be fixed for 1.6.0 final.

**Comment by aanand** [comment]
Fix is in #2722.
